### PR TITLE
JavaScript functions, when passed to Objective-C as blocks, will now round-trip back to JavaScript, as the initial function object.

### PR DESCRIPTION
**Pull request by PanayotCankov**

### CODE REVIEW

File: src/NativeScript/ObjC/Block/ObjCBlockType.mm [hunk 1]
```diff
@@ -88,7 +96,14 @@ static void disposeBlock(JSBlock* block) {
 JSValue ObjCBlockType::read(ExecState* execState, const void* buffer, JSCell* self) {
     GlobalObject* globalObject = jsCast<GlobalObject*>(execState->lexicalGlobalObject());
     ObjCBlockType* blockType = jsCast<ObjCBlockType*>(self);
-    return ObjCBlockCall::create(execState->vm(), globalObject->objCBlockCallStructure(), *static_cast<const id*>(buffer), blockType);
+    id block = *static_cast<const id*>(buffer);
+
+    JSValue objCBlockCallback = JSBlock::getJSFunction(block);
```
Comment (jasssonpet) [inline src/NativeScript/ObjC/Block/ObjCBlockType.mm]: You can put the assignment in the if statement to reduce its scope.

File: tests/TestFixtures/TNSTestNativeCallbacks.h [hunk 2]
```diff
@@ -12,6 +12,10 @@
 
 @interface TNSTestNativeCallbacks : NSObject
 
+typedef void (^TNSBlock)();
+
+@property(assign, nonatomic) TNSBlock block;
```
Comment (jasssonpet) [inline tests/TestFixtures/TNSTestNativeCallbacks.h]: I would suggest to move this to `TNSObjCTypes.m` where we have tests like:
* `methodWithNSDate`
* `methodWithNSArray`
* `methodWithNSDictionary`
* `methodWithNSData`

which test the same thing (that the object is the same).

Also removing the property from the header won't make the test fail, an is a bit brittle.

Comment (fealebenpae) [inline tests/TestFixtures/TNSTestNativeCallbacks.h]: > imo JS is brittle.

FTFY :grinning: 

We can add an expectation that the `block` property exists with the `in` operator, but what you've touched upon is a problem everywhere, not just in this test.

Comment (jasssonpet) [inline tests/TestFixtures/TNSTestNativeCallbacks.h]: That's why we try to avoid accessing native properties in our tests.

**Comment by fealebenpae** [comment]
run ci

**Comment by fealebenpae** [comment]
The build currently fails with:
`src/NativeScript/ObjC/Block/ObjCBlockType.mm:78:28: error: cannot cast from type 'id' to pointer type 'NativeScript::JSBlock *'`
`reinterpret_cast` should fix it.

**Pull request merged**